From 0defb9078460cb18a1c4e396d30b539103069846 Mon Sep 17 00:00:00 2001
From: Lukas Tribus <luky-37@hotmail.com>
Date: Wed, 13 Feb 2013 23:35:39 +0100
Subject: DOC: tfo: bump required kernel to linux-3.7

Support for server side TFO was actually introduced in linux-3.7,
linux-3.6 just has client support.

This patch fixes documentation and a code comment about the
kernel requirement. It also fixes a wrong tfo related code
comment in src/proto_tcp.c.
---
 doc/configuration.txt    |    2 +-
 include/types/listener.h |    2 +-
 src/proto_tcp.c          |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/doc/configuration.txt b/doc/configuration.txt
index 49e64c5..70a5188 100644
--- a/doc/configuration.txt
+++ b/doc/configuration.txt
@@ -7287,7 +7287,7 @@ strict-sni
   See the "crt" option for more information.
 
 tfo
-  Is an optional keyword which is supported only on Linux kernels >= 3.6. It
+  Is an optional keyword which is supported only on Linux kernels >= 3.7. It
   enables TCP Fast Open on the listening socket, which means that clients which
   support this feature will be able to send a request and receive a response
   during the 3-way handshake starting from second connection, thus saving one
diff --git a/include/types/listener.h b/include/types/listener.h
index 2f584e3..6e93125 100644
--- a/include/types/listener.h
+++ b/include/types/listener.h
@@ -89,7 +89,7 @@ enum {
 #define LI_O_CHK_MONNET 0x0020  /* check the source against a monitor-net rule */
 #define LI_O_ACC_PROXY  0x0040  /* find the proxied address in the first request line */
 #define LI_O_UNLIMITED  0x0080  /* listener not subject to global limits (peers & stats socket) */
-#define LI_O_TCP_FO     0x0100  /* enable TCP Fast Open (linux >= 3.6) */
+#define LI_O_TCP_FO     0x0100  /* enable TCP Fast Open (linux >= 3.7) */
 #define LI_O_V6ONLY     0x0200  /* bind to IPv6 only on Linux >= 2.4.21 */
 #define LI_O_V4V6       0x0400  /* bind to IPv4/IPv6 on Linux >= 2.4.21 */
 
diff --git a/src/proto_tcp.c b/src/proto_tcp.c
index 60b111c..6bd5761 100644
--- a/src/proto_tcp.c
+++ b/src/proto_tcp.c
@@ -1776,7 +1776,7 @@ static int bind_parse_defer_accept(char **args, int cur_arg, struct proxy *px, s
 #endif
 
 #ifdef TCP_FASTOPEN
-/* parse the "defer-accept" bind keyword */
+/* parse the "tfo" bind keyword */
 static int bind_parse_tfo(char **args, int cur_arg, struct proxy *px, struct bind_conf *conf, char **err)
 {
 	struct listener *l;
-- 
1.7.1

